Library/Formula/opencoarrays.rb

```diff
@@ -0,0 +1,85 @@
+class Opencoarrays < Formula
+  desc "open-source coarray Fortran ABI, API, and compiler wrapper"
+  homepage "http://opencoarrays.org"
+  url "https://github.com/sourceryinstitute/opencoarrays/releases/download/1.2.1/opencoarrays-1.2.1.tar.gz"
+  sha256 "2771a8f6d44f6492d0b30731ab7b7803bfea1a14aab4f6936ddd2a400956e2bb"
+
+  head "https://github.com/sourceryinstitute/opencoarrays.git"
+
+  option "without-test", "Skip build time tests (requires properly configured MPICH and firewall settings)"
+
+  depends_on :fortran
+  depends_on :mpi => [:cc, :f90]
+  depends_on "cmake" => :build
+
+  # GCC >= 5.2 required; no other Homebrew supported compilers are compatible
+  # but the mpicc compiler may be clang (which is ABI compatible with GCC)
+  fails_with :llvm
+  fails_with :gcc
+  fails_with :gcc_4_0
+  ("4.3".."4.9").each do |series|
+    fails_with :gcc => series
+  end
+  fails_with :gcc => "5.1"
+
+  def install
+    mkdir "build" do
+      ENV["FC"] = "mpif90"
+      ENV["CC"] = "mpicc"
+      # Release build type in std_cmake_args can break the build due to GCC bugs
+      # at certain optimization levels. See sourceryinstitute/opencoarrays#28
+      # for further details
+      cmake_args = std_cmake_args
+      cmake_args.delete("-DCMAKE_BUILD_TYPE=Release")
+      system "cmake", "..", *cmake_args
+      system "make"
+      # see sourceryinstitute/opencoarrays/blob/master/STATUS.md#compiler-issues-gnu
+      gfortran_version = /\d\.\d+/.match(`mpif90 --version`)
+      test_excludes = ["-E", "co_reduce"] if gfortran_version.to_s < "5.3"
+      system "ctest", *test_excludes if build.with? "test"
```

Comment:
> ah, got it... interesting, in my testing (manually changing/setting values) no problems arose due to that issue.

Just to clarify: That's because newer Ruby versions (OS X 10.9+) do the same thing for both `*[]` and `*nil`. But we try not to break things for Ruby 1.8 users and there the behavior is different.